File: src/main/java/org/lucidfox/questfiller/model/npc/NPCQuest.java
package org.lucidfox.questfiller.model.npc;

public final class NPCQuest implements Comparable<NPCQuest> {
	private final String name;
	private final boolean starts;
	private final boolean finishes;
	
	public NPCQuest(String name, boolean starts, boolean finishes) {
		this.name = name;
		this.starts = starts;
		this.finishes = finishes;
	}

	public String getName() {
		return name;
	}

	public boolean isStarts() {
		return starts;
	}

	public boolean isFinishes() {
		return finishes;
	}

	@Override
	public int compareTo(NPCQuest o) {
		return name.compareTo(o.name);
	}
}

File: src/model/npc/NPCQuest.js
export default class NPCQuest {
	/**
	 * 
	 * @param {string} name 
	 * @param {boolean} starts 
	 * @param {boolean} finishes 
	 */
	constructor(name, starts, finishes) {
		this.name = name;
		this.starts = starts;
		this.finishes = finishes;
	}

	/**
	 * 
	 * @param {NPCQuest} other 
	 */
	compareTo(other) {
		return this.name.localeCompare(other.name);
	}
}
